src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final String COLOR = new ColorSupplier().getRandomColor();
+    private static final int LIMIT_NUMBER = 20;
+
+    public Figure getRandomFigure() {
+        switch (RANDOM.nextInt(4)) {
+            case 0:
+                return new Circle(COLOR, RANDOM.nextInt(LIMIT_NUMBER));
+            case 1:
+                return new Square(COLOR, RANDOM.nextInt(LIMIT_NUMBER));
+            case 2:
+                return new IsoscelesTrapezoid(COLOR,
+                        RANDOM.nextInt(LIMIT_NUMBER),
+                        RANDOM.nextInt(LIMIT_NUMBER),
+                        RANDOM.nextInt(LIMIT_NUMBER));
+            case 3:
+                return new Rectangle(COLOR,
+                        RANDOM.nextInt(LIMIT_NUMBER),
+                        RANDOM.nextInt(LIMIT_NUMBER));
+            case 4:
+                return new RightTriangle(COLOR,
+                        RANDOM.nextInt(LIMIT_NUMBER),
+                        RANDOM.nextInt(LIMIT_NUMBER));
+            default:
+                return new Circle("white", 10);

```

Comment:
Don't return `null` from a method. Instead of having a default case that returns a new object, you should ensure that all possible cases are covered in the switch statement.